src/dicomweb_client/api.py

```diff
@@ -1357,16 +1411,53 @@ def _http_post(
 
         '''
         logger.debug('POST: {} {}'.format(url, headers))
-        response = self._session.post(url=url, data=data, headers=headers)
+
+        def serve_data_chunks(data):
+            for i, offset in enumerate(range(0, len(data), self._chunk_size)):
+                end = offset + self._chunk_size
+                yield data[offset:end]
+
+        if self._chunk_size is not None and len(data) > self._chunk_size:
+            logger.info('store data in chunks using chunked transfer encoding')
+            chunked_headers = dict(headers)
+            chunked_headers['Transfer-Encoding'] = 'chunked'
+            chunked_headers['Cache-Control'] = 'no-cache'
+            chunked_headers['Connection'] = 'Keep-Alive'
+            data_chunks = serve_data_chunks(data)
+            response = self._session.post(
+                url=url,
+                data=data_chunks,
+                headers=headers
+            )
+        else:
+            response = self._session.post(url=url, data=data, headers=headers)
         logger.debug('request status code: {}'.format(response.status_code))
-        response.raise_for_status()
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as error:
+            raise HTTPError(error)
+        except requests.exceptions.ConnectionError as error:
+            raise HTTPError(error[0])
+        if not response.ok:
+            logger.warning('storage was not successful for all instances')
+            payload = response.content
+            tree = ET.fromstring(payload)
+            dataset = _load_xml_dataset(tree)
```

Comment:
In `_http_post_multipart_application_dicom`, when the `response.content` returned by `_http_post` is parsed, `Content-Type` is checked to decide whether it should be loaded as `json` or `xml`. Here, in case of a response with an error status code, content is assumed to be `xml`. Is it not possible for the content of an error response to be in `json`?

Reply:
See #26